task_reactredux/task1/src/reducers/MailReducer.js

```diff
@@ -0,0 +1,26 @@
+import { validateMail } from '../Validation';
+import { ON_MAIL_CHANGE, VALIDATE_MAIL } from '../Constants';
+
+const initialState = {
+    mail: '',
+    mailValid: false
+};
+
+const MailReducer = (state = initialState, action) => {
+    switch (action.type) {
+        case ON_MAIL_CHANGE:
+            return {
+                ...state,
+                mail: action.mail
+            }
+        case VALIDATE_MAIL:
```

Comment:
Do you really need 2 actions for this? As far as i can see, you are calling this actions only together 
```javsacript
    this.props.onPasswordChange(value);
    this.props.validatePassword(value);
    //
    this.props.onMailChange(value);
    this.props.validateMail(value);
```
Probably you could merge them.
```javascript
        case ON_MAIL_CHANGE:
            return {
                ...state,
                mail: action.mail,
                mailValid: validateMail(action.mail)
            }
```